lib/ansible/modules/network/onyx/onyx_syslog_remote.py

```diff
@@ -0,0 +1,346 @@
+                if host in self._current_config:
+                    self._current_config[host].update({'filter': filter_type, 'filter_str': filter_str})
+                else:
+                    self._current_config[host] = dict(filter=filter_type, filter_str=filter_str)
+
+    def get_required_config(self):
+        self._required_config = dict()
+        required_config = dict()
+        module_params = self._module.params
+        port = module_params.get('port')
+        trap = module_params.get('trap')
+        trap_override = module_params.get('trap_override')
+        filtered = module_params.get('filter')
+
+        required_config['host'] = module_params.get('host')
+        required_config['enabled'] = module_params.get('enabled')
+
+        if port:
+            required_config['port'] = port
+        if trap:
+            required_config['trap'] = trap
+        if trap_override:
+            required_config['trap_override'] = trap_override
+        if filtered:
+            required_config['filter'] = filtered
+            required_config['filter_str'] = module_params.get('filter_str', '')
+
+        self.validate_param_values(required_config)
+        self._required_config = required_config
+
+    def generate_commands(self):
+        required_config = self._required_config
+        current_config = self._current_config
+        host = required_config.get('host')
+        '''
+        cases:
+        if host in current config and current config != required config and its enable
+        if host in current config and current config == required config and its disable
+        if host not in current config and its enable
+        '''
+        host_config = current_config.get(host, dict())
+        if 'port' in required_config:
+            if required_config['enabled']:
+                if required_config['port'] != host_config.get('port', None) or not host_config:
+                    '''Edit/Create new one'''
+                    self._commands.append('logging {0} port {1}'.format(host, required_config['port']))
+
+            elif 'port' in host_config:
+                self._commands.append('no logging {0} port'.format(host))
+
+        if 'trap' in required_config or 'trap_override':
+            trap_commands = self._get_trap_commands(host)
+            self._commands += trap_commands
+
+        if 'filter' in required_config:
+            is_change = host_config.get('filter', None) != required_config['filter'] or \
+                host_config.get('filter_str', None) != required_config['filter_str']
+            if required_config['enabled']:
+                if is_change or not host_config:
+                    self._commands.append('logging {0} filter {1} {2}'.format(host, required_config['filter'], required_config['filter_str']))
+
+            else:
+                if 'filter' in host_config:
+                    self._commands.append('no logging {0} filter'.format(host))
+
+        if len(required_config) == 2:
```

Comment:
this is not a good test, and could cause bugs in the future

Reply:
Here I must make sure that there is no other keys but [enabled, host] which are required, So you prefer to add a flag and in each check for other keys set the flag to False?